modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java

```diff
@@ -211,8 +206,7 @@ public Neo4jContainer withEnterpriseEdition() {
                 String.format("Cannot use enterprise version with alternative image %s.", getDockerImageName())
             );
         }
-
-        setDockerImageName(DEFAULT_IMAGE_NAME.withTag(ENTERPRISE_TAG).asCanonicalNameString());
+        setDockerImageName(DockerImageName.parse(getDockerImageName() + ENTERPRISE_SUFFIX).asCanonicalNameString());
```

Comment:
this will break existing users. Why not improve the exception's message?

Reply:
Will play with this. Thanks for the feedback.